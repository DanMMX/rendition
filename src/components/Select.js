import * as React from 'react'
import styled, { withTheme } from 'styled-components'
import asRendition from '../asRendition'
import { px } from '../utils'
import { radius } from '../theme'
import { compose } from 'recompose'

const Base = styled.select`
  border-radius: ${px(radius)};
  height: ${props =>
    px(props.emphasized ? props.theme.space[5] : props.theme.space[4])};
  font-size: inherit;
  border: 1px solid ${props => props.theme.colors.gray.main};
  padding-top: 0px;
  padding-bottom: 0px;
  padding-left: 16px;
  padding-right: ${props =>
    px(props.emphasized ? props.theme.space[5] : props.theme.space[4])};
  background-color: white;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  width: 100%;
  min-width: 90px;

  &:hover {
    box-shadow: 0 0 4px 1px rgba(0, 0, 0, 0.1);
  }
`

const Wrapper = styled.span`
  display: inline-block;
  position: relative;

  &::after {
    content: "";
    width: 0;
    height: 0;
    border-left: 4px solid transparent;
    border-right: 4px solid transparent;
    border-top: 5px solid ${props => props.theme.colors.gray.dark};

    position: absolute;
    right: 16px;
    top: ${props => px(props.emphasized ? 20 : 16)};
  }
`

const Component = ({ emphasized, children, value, onChange, ...props }) => {
  return (
    <Wrapper emphasized={emphasized} {...props}>
      <Base
        emphasized={emphasized}
        value={value}
        onChange={onChange}
        children={children}
      />
    </Wrapper>
  )
}

export default compose(withTheme, asRendition)(Component)
