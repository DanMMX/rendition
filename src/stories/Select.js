import { h } from 'preact'
import { storiesOf } from '@storybook/react'
import styled from 'styled-components'
import Select from '../components/Select'

const Container = styled.div`margin: 30px;`

storiesOf('Select', module)
  .addWithInfo('Standard', () => {
    return (
      <Container>
        <Select>
          <option>Option 1</option>
          <option>Option 2</option>
          <option>Option 3</option>
        </Select>
      </Container>
    )
  })
  .addWithInfo('Emphasized', () => {
    return (
      <Container>
        <Select emphasized>
          <option>Option 1</option>
          <option>Option 2</option>
          <option>Option 3</option>
        </Select>
      </Container>
    )
  })
