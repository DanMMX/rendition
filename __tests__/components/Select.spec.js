import React from 'react';
import renderer from 'react-test-renderer';
import Provider from '../../src/components/Provider'
import Select from '../../src/components/Select'

test('Select renders correctly', () => {
  const component = renderer.create(
    <Provider>
      <Select>
        <option value={1}>Option 1</option>
        <option value={2}>Option 2</option>
        <option value={3}>Option 3</option>
      </Select>
    </Provider>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
