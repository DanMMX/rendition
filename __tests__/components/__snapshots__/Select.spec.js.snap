// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select renders correctly 1`] = `
<div
  className="sc-bdVaJa jKopLr"
>
  <span
    className="sc-dnqmqq gDpXJY sc-htoDjs gdpVxp"
  >
    <select
      className="sc-gzVnrw ePUKPC"
      onChange={undefined}
      value={undefined}
    >
      <option
        value={1}
      >
        Option 1
      </option>
      <option
        value={2}
      >
        Option 2
      </option>
      <option
        value={3}
      >
        Option 3
      </option>
    </select>
  </span>
</div>
`;
